New resource `azurerm_virtual_machine_configuration_policy_assignment` (#11334)
terraform-providers/terraform-provider-azurerm: website/docs/r/virtual_machine_configuration_policy_assignment.html.markdown

---
subcategory: "Policy"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_virtual_machine_configuration_policy_assignment"
description: |-
  Manages a Virtual Machine Configuration Policy Assignment.
---

# azurerm_virtual_machine_configuration_policy_assignment

> Manages a Virtual Machine Configuration Policy Assignment.

Reviewer: we're not "managing" something here so much as
```suggestion
Applies a Configuration Policy to a Virtual Machine.
```
?

  }

  source_image_reference {
    publisher = "MicrosoftWindowsServer"
    offer     = "WindowsServer"
    sku       = "2016-Datacenter"
    version   = "latest"
  }
}

resource "azurerm_virtual_machine_configuration_policy_assignment" "example" {
  name               = "example-gca"
  location           = azurerm_linux_virtual_machine.example.id
  virtual_machine_id = azurerm_linux_virtual_machine.example.id
  guest_configuration {
    name    = "WhitelistedApplication"
    version = "1.*"

    parameter {
      name  = "[InstalledApplication]bwhitelistedapp;Name"
      value = "NotePad,sql"
    }
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name of the Virtual Machine Configuration Policy Assignment. Changing this forces a new resource to be created.

* `location` - (Required) The Azure location where the Virtual Machine Configuration Policy Assignment should exist. Changing this forces a new resource to be created.

* `virtual_machine_id` - (Required) The resource ID of the Virtual Machine which this Guest Configuration Assignment should apply to. Changing this forces a new resource to be created.

---

> * `guest_configuration` - (Required)  A `guest_configuration` block as defined below.

Reviewer: this doesn't match schema.

given policy is in the name, and policy_assingment.policy is an inline policy not a policy id that is assigned, maybe policy doesn't make the most sens

configuration_checks? checks?
```
configuration {
  parameter {
    name = "someprop"
    matches = "somevalue"
  }
}
```

WDYT? the schema and names here are confusing and all over and i'm not sure what the best UX here would be

Author: You are absolutely correct! I forgot to change this after I changed the schema. I just fixed this.

---

An `guest_configuration` block supports the following:

* `name` - (Required) The name of the Guest Configuration that will be assigned in this Guest Configuration Assignment.

> * `parameter` - (Optional) One or more `parameter` blocks as defined below.

Reviewer: ```suggestion
* `parameter` - (Optional) One or more `parameter` blocks which define what configuration parameters and values againsts.
```

* `version` - (Optional) The version of the Guest Configuration that will be assigned in this Guest Configuration Assignment.

---

An `parameter` block supports the following:

> * `name` - (Required) The name of the configuration parameter.
> 
> * `value` - (Required) The value of the configuration parameter.

Reviewer: could we be more clear in these docs
```suggestion
* `name` - (Required) The name of the configuration parameter to check.

* `value` - (Required) The value to check the configuration parameter with.
```
